resolwe_bio/processes/clustering/hierarchical_clustering.yml

```diff
@@ -124,9 +117,7 @@
     runtime: polyglot
     language: bash
     program: |
-
       {% for e in exps %}
-
         {% if e.source != (exps|first).source %}
           re-warning "All expression data must be annotated by the same genome database."
           re-error "Sample {{ exps|first|name }} has {{ (exps|first).source }} gene IDs, while sample {{ e|name }} has {{ e.source }} gene IDs."
```

Comment:
I know this is not part of your PR, but what is logic behind this? We have a warning and two errors directly afterwards. Why raise a warning if error follows immediately? And why raise two errors, second of which is totally unrelated to the if clause that triggers it...

Reply:
Same holds for many error / warning messages all over this process.

Reply:
True, my apologies. I agree: we should use one verbose message and `re-error`.